src/lib/auth/AUTH0.ts

```diff
@@ -0,0 +1,85 @@
+import type { AuthData } from '$lib/models/AuthProvider';
+import AuthProvider from '$lib/models/AuthProvider';
+import { browser } from '$app/environment';
+import { login } from '$lib/stores/User';
+import type { User } from '$lib/models/User';
+import * as api from '$lib/api';
+import auth0 from 'auth0-js';
+
+interface Auth0Data extends AuthData {
+  clientid: string;
+  connection: string;
+}
+
+class Auth0 extends AuthProvider implements Auth0Data {
+  clientid: string;
+  connection: string;
+
+  constructor(data: Auth0Data) {
+    super(data);
+    this.clientid = data.clientid;
+    this.connection = data.connection;
+  }
+
+  //TODO: create real return types
```

Comment:
idk... I wanted to talk about that. I just didn't really like the structure of all the code like this stuff:
```
if (newUser?.token) {
          login(newUser.token);
          return false;
        } else {
          return true;
        }
      } catch (error) {
        return true;
      }
    }
    return true;

```
Looks like this can be done cleaner